x-pack/metricbeat/module/aws/aws.go

```diff
@@ -56,6 +56,22 @@ type Tag struct {
 	Value string `config:"value"`
 }
 
+type Op string
+
+const (
+	Leaf Op = "leaf"
+	And  Op = "and"
+	Or   Op = "or"
+	Not  Op = "not"
+)
+
+type Rule struct {
+	Op    Op    `config:"op"`
+	Left  *Rule `config:"left"`
```

Comment:
I seem to hit an infinite loop when trying to parse this type of configuration using elastic/go-ucfg. is this feasible to support?

Reply:
perhaps something like https://github.com/go-extras/tahwil can be leveraged